client/core/core.go

```diff
@@ -2881,6 +2883,45 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 	c.updateBalances(counts)
 }
 
+func (c *Core) PromptShutdown() bool {
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+	ok := true
+	for _, dc := range c.conns {
+		if dc.hasActiveOrders() {
+			ok = false
+			break
+		}
+	}
+	if !ok {
+		fmt.Print("You have active orders. Shutting down now may result in failed swaps and account penalization. " +
+			"Do you want to quit anyway? ('y' to quit, enter to abort):")
```

Comment:
Note that we are not preventing new orders while answering this prompt since `Trade`releases `c.connMtx` after looking up the `dexConnection`, and it it could still be working up to a `newTrackedTrade` while this function checks for active orders.  There's not an obvious way to avoid this race so I'm not requesting a resolution, but let's make a note of it.